grant-project/grant-monorepo
# Unify markdown, make styles consistent

Closes #21. Relies on #35.

### What This Does

* Switches all markdown engines to Showdown which is way more lenient than `remark` (What `react-markdown` used.)
* Creates one unified style for markdown, making proposal preview and the actual view identical.
* Adds xss filtering to markdown using the `xss` module

### Steps to Test

* Create a new proposal
* Use my [markdown stress tester](https://gist.github.com/wbobeirne/8db15ca3fdf9459002dcd8144ffa4281) for description
* Confirm the preview looks good.
* Create the proposal, confirm the proposal view looks good and identical to preview.

### Known issues

* Tables and images don't support _all_ of Github's fancy markdown.
* Syntax highlighting for language-specified code doesn't work (do we want this?)
